crates/cosmos/cosmos-chain-components/src/impls/queries/write_ack_event.rs

```diff
@@ -1,65 +1,88 @@
 use ibc_relayer_types::core::ics04_channel::packet::Packet;
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
 use ibc_relayer_types::events::IbcEvent;
+use ibc_relayer_types::events::WithBlockDataType;
+use ibc_relayer_types::Height;
+use tendermint_rpc::{Client, Error as TendermintRpcError};
 
-use crate::traits::chain_handle::HasBlockingChainHandle;
+pub struct QueryWriteAckEventFromAbci;
 
-pub struct QueryWriteAckEventFromChainHandle;
-
-impl<Chain, Counterparty> WriteAckQuerier<Chain, Counterparty> for QueryWriteAckEventFromChainHandle
+impl<Chain, Counterparty> WriteAckQuerier<Chain, Counterparty> for QueryWriteAckEventFromAbci
 where
-    Chain: HasWriteAckEvent<Counterparty, WriteAckEvent = WriteAcknowledgement>
+    Chain: HasErrorType
+        + HasWriteAckEvent<Counterparty, WriteAckEvent = WriteAcknowledgement>
         + HasIbcPacketTypes<Counterparty, IncomingPacket = Packet>
-        + HasBlockingChainHandle,
+        + HasRpcClient
+        + HasGrpcAddress
+        + HasComponents
+        + CanRaiseError<TendermintRpcError>
+        + CanRaiseError<Ics04Error>
+        + CanRaiseError<Ics02Error>
+        + CanRaiseError<tonic::transport::Error>
+        + CanRaiseError<ibc_relayer::error::Error>,
 {
     async fn query_write_ack_event(
         chain: &Chain,
         packet: &Packet,
     ) -> Result<Option<Chain::WriteAckEvent>, Chain::Error> {
-        let packet = packet.clone();
+        let rpc_client = chain.rpc_client();
+
+        let abci_info = rpc_client.abci_info().await.map_err(Chain::raise_error)?;
 
-        chain
-            .with_blocking_chain_handle(move |chain_handle| {
-                let status = chain_handle
-                    .query_application_status()
-                    .map_err(Chain::raise_error)?;
+        let response = rpc_client
+            .header(abci_info.last_block_height)
+            .await
+            .map_err(Chain::raise_error)?;
 
-                let query_height = Qualified::Equal(status.height);
+        let query_height = Height::new(
+            ChainId::chain_version(response.header.chain_id.as_str()),
+            u64::from(abci_info.last_block_height),
+        )
+        .map_err(Chain::raise_error)?;
 
-                let path_ident = PathIdentifiers {
-                    port_id: packet.destination_port.clone(),
-                    channel_id: packet.destination_channel.clone(),
-                    counterparty_port_id: packet.source_port.clone(),
-                    counterparty_channel_id: packet.source_channel.clone(),
-                };
+        let packet = packet.clone();
+        let request = QueryPacketEventDataRequest {
+            event_id: WithBlockDataType::WriteAck,
+            source_channel_id: packet.destination_channel.clone(),
+            source_port_id: packet.destination_port.clone(),
+            destination_channel_id: packet.source_channel.clone(),
+            destination_port_id: packet.source_port.clone(),
```

Comment:
> i'm still not sure which one is the source and which one is the dest data.

Hmm you are right. It looks like the source should match the packet source, and vice versa for destination.

Actually looking at the code, it looks like this component is not used anywhere in the relayer code, other than the mock relayer. So I think we can remove this component (`QueryWriteAckEventFromAbci`) altogether, if it does not introduce any compile error.